src/operator/custom/custom-inl.h

```diff
@@ -139,38 +143,51 @@ class CustomOperator {
       destructing_ = true;
       cv_.notify_all();
     }
-    worker_.join();
+    for (auto &worker : workers_)
+      worker.join();
```

Comment:
Have you evaluated this when the custom op is part of a bigger graph and if there is any performance impact ? Since the CustomOperator is static its lifetime is till the end of the program and destructor of customoperator gets called at the end. This means there is one thread that is waiting on the condition variable while other 11 threads tryign to obtain the lock and in a blocked state. Since these are idle threads, I am not sure if the impact will be significant but good to verify. Will also help us come up with a good default for MXNET_CUSTOM_OP_NUM_THREADS

Reply:
Thank you!

Sorry that I do not have any server with multiple GPUs to evaluate a big graph.

In this PR, the number of threads will increase when threads are not enough, and the maximum number id MXNET_CUSTOM_OP_NUM_THREADS.

There are always threads which get the lock and execute the operator function, so I think the idle threads do not drop the performance.

Reply:
Yes it's fine during op execution but even after custom op execution there is one thread waiting on CV and other 11 trying to acquire lock.

Reply:
I see. 
I ever thought that decrease the number of threads when threads are idle, however it is difficult to estimate the number.

Reply:
can you run the example/reinforcement-learning/dqn  which includes a custom op on CPU to check for performance.

Reply:
IT may be not available to check for performance on only CPU，since there Is only a computational stream usually. I will find a server with multiple GPUs and check on it.

Reply:
Yes this change is mainly for performance improvements on a GPU machine, but we should be careful not to impact cpu performance for inference. Otherwise we should keep the default small.

Reply:
Sorry that I met the problem when running the RL demo:
```bash
(gdb) bt                                                                                                                                    
#0  0x00007ffff7e1ff73 in free () at /usr/lib/libc.so.6                                                                                     
#1  0x00007fffbf2c9ad1 in ALEInterface::welcomeMessage[abi:cxx11]() () at /usr/lib/python2.7/site-packages/ale_python_interface/libale_c.so 
#2  0x00007fffbf2c9d0a in ALEInterface::ALEInterface() () at /usr/lib/python2.7/site-packages/ale_python_interface/libale_c.so              
#3  0x00007fffbf2c7fbb in ALE_new () at /usr/lib/python2.7/site-packages/ale_python_interface/libale_c.so                                   
#4  0x00007ffff6d236d0 in ffi_call_unix64 () at /usr/lib/libffi.so.6                                                                        
#5  0x00007ffff6d230a0 in ffi_call () at /usr/lib/libffi.so.6                                                                               
#6  0x00007ffff6cc9d77 in _ctypes_callproc () at /usr/lib/python2.7/lib-dynload/_ctypes.so                                                 
#7  0x00007ffff6cc37d0 in  () at /usr/lib/python2.7/lib-dynload/_ctypes.so                                                                 
#8  0x00007ffff7c8cb43 in PyObject_Call () at /usr/lib/libpython2.7.so.1.0                                                                  
#9  0x00007ffff7c1c10e in PyEval_EvalFrameEx () at /usr/lib/libpython2.7.so.1.0                                                            
#10 0x00007ffff7c156ba in PyEval_EvalCodeEx () at /usr/lib/libpython2.7.so.1.0             
```

Reply:
I'm checking it.

Reply:
In my experiment, the performance does not drop in DQN example. The FPS keeps 1500 on my laptop with only CPU i7-7500U(2c4t).